### PR TITLE
Update CHANGELOG for Packer 1.11.0-beta

### PR DESCRIPTION
**Pull request by nywilken**
This change simply updates the CHANGELOG with updates to 1.11.0, since the last alpha2 release. 

Depends on: #12911
Depends on: #12915
Depends on: #12936

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -50,6 +62,22 @@
...
+     install plugins manually using `packer plugins install --path`.
+     [GH-12911](https://github.com/hashicorp/packer/pull/12911)
+* core: Remote plugins installed containing an internal version number that
+     differs from the version number within the binary name can lead to
+     confusion when tracking Packer plugin version information. To help track
+     such deprecates in the plugin version, `packer init` and `packer plugin
+     install` have been updated to install such plugins as dev versions (e.g.
+     1.0.0-dev). Users are encouraged to notify plugin maintainers of any
+     version mismatches.
+     [GH-12915](https://github.com/hashicorp/packer/pull/12915)
```
Comment (nywilken) [inline CHANGELOG.md:79]:
```suggestion
     [GH-12915](https://github.com/hashicorp/packer/pull/12915), [GH-12953](https://github.com/hashicorp/packer/pull/12953)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -50,6 +62,22 @@
      command. This addition makes it possible to pipe commands like
      `packer plugins installed` with it for speedy cleanup of installed plugins.
      [GH-12886](https://github.com/hashicorp/packer/pull/12886)
+* core: Relax Packer source address URIs within the `required_plugins` block to
+     support the installation of local plugin binary using a custom or internal
+     source address (e.g. mycompay.com/plugins/happyorg/happycloud). Remote
+     installation using `packer init` or `packer plugins install`  does not
+     support non-GitHub source URIs. Users using internal source URIs must
+     install plugins manually using `packer plugins install --path`.
+     [GH-12911](https://github.com/hashicorp/packer/pull/12911)
```
Comment (nywilken) [inline CHANGELOG.md:71]:
```suggestion
     [GH-12911](https://github.com/hashicorp/packer/pull/12911), [GH-12962](https://github.com/hashicorp/packer/pull/12962)
```
```suggestion
     [GH-12911](https://github.com/hashicorp/packer/pull/12911)
```
```suggestion
     [GH-12911](https://github.com/hashicorp/packer/pull/12911)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -50,6 +62,22 @@
...
+* core: Remote plugins installed containing an internal version number that
+     differs from the version number within the binary name can lead to
+     confusion when tracking Packer plugin version information. To help track
+     such deprecates in the plugin version, `packer init` and `packer plugin
+     install` have been updated to install such plugins as dev versions (e.g.
+     1.0.0-dev). Users are encouraged to notify plugin maintainers of any
+     version mismatches.
+     [GH-12915](https://github.com/hashicorp/packer/pull/12915), [GH-12953](https://github.com/hashicorp/packer/pull/12953)
```
Comment (nywilken) [inline CHANGELOG.md:79]:
```suggestion
     [GH-12915](https://github.com/hashicorp/packer/pull/12915), [GH-12953](https://github.com/hashicorp/packer/pull/12953), [GH-12972](https://github.com/hashicorp/packer/pull/12972)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -50,6 +62,22 @@
      command. This addition makes it possible to pipe commands like
      `packer plugins installed` with it for speedy cleanup of installed plugins.
      [GH-12886](https://github.com/hashicorp/packer/pull/12886)
+* core: Relax Packer source address URIs within the `required_plugins` block to
+     support the installation of local plugin binary using a custom or internal
+     source address (e.g. mycompay.com/plugins/happyorg/happycloud). Remote
```
Comment (lbajolet-hashicorp) [inline CHANGELOG.md:67]: Nit:

```suggestion
     source address (e.g. mycompany.com/plugins/happyorg/happycloud). Remote
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -50,6 +62,22 @@
...
+     installation using `packer init` or `packer plugins install`  does not
+     support non-GitHub source URIs. Users using internal source URIs must
```
Comment (lbajolet-hashicorp) [inline CHANGELOG.md:69]: Just to make it clear that any non-Github URI can use this method, not just internal hosts

```suggestion
     support non-GitHub source URIs. Users using alternative hosts must
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -50,6 +62,22 @@
...
+     install plugins manually using `packer plugins install --path`.
+     [GH-12911](https://github.com/hashicorp/packer/pull/12911)
+* core: Remote plugins installed containing an internal version number that
+     differs from the version number within the binary name can lead to
+     confusion when tracking Packer plugin version information. To help track
+     such deprecates in the plugin version, `packer init` and `packer plugin
```
Comment (lbajolet-hashicorp) [inline CHANGELOG.md:75]:
```suggestion
     such discrepancies in the plugin version, `packer init` and `packer plugin
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -50,6 +62,22 @@
...
+     install` have been updated to install such plugins as dev versions (e.g.
```
Comment (lbajolet-hashicorp) [inline CHANGELOG.md:76]: That's no longer true is it? We reject these installations with an explicit error.

```suggestion
     install` have been updated to reject installation of such plugins.
```

**Review by lbajolet-hashicorp — COMMENTED** [review]
Couple nits/suggestions, but overall LGTM!

**Comment by nywilken** [comment]
@lbajolet-hashicorp if this is good to go can you give me approval to unblock the merge. 🙏🏽

**Review by lbajolet-hashicorp — APPROVED** [review]
Ah yeah I forgot to approve it sorry!

**Pull request merged**